diff --git a/config/rv32gc/config.vh b/config/rv32gc/config.vh
index 890bc5208..c22b5f0e5 100644
--- a/config/rv32gc/config.vh
+++ b/config/rv32gc/config.vh
@@ -3,6 +3,7 @@
 //
 // Written: David_Harris@hmc.edu 4 January 2021
 // Modified: Jordan Carlin jcarlin@hmc.edu 14 May 2024
+//           Giovanni Sirtori <gs86@rice.edu> (HYDRA integration), 3 May 2026 
 //
 // Purpose: Specify which features of Wally are enabled and set
 //          configuration parameters
@@ -188,6 +189,13 @@ localparam logic [63:0] SDC_RANGE        = 64'h00000FFF;
 localparam logic SPI_SUPPORTED = 1;
 localparam logic [63:0] SPI_BASE         = 64'h10040000;
 localparam logic [63:0] SPI_RANGE        = 64'h00000FFF;
+localparam logic HYDRA_SUPPORTED            = 1;
+localparam logic [63:0] HYDRA_BASE          = 64'h2000_0000;
+localparam logic [63:0] HYDRA_RANGE         = 64'h0040_FFFF;
+localparam logic [63:0] HYDRA_MMR_BASE      = 64'h2000_0000;
+localparam logic [63:0] HYDRA_MMR_RANGE     = 64'h0000_0FFF;
+localparam logic [63:0] HYDRA_SCRATCH_BASE  = 64'h2001_0000;
+localparam logic [63:0] HYDRA_SCRATCH_RANGE = 64'h003F_FFFF;
 
 // Bus Interface width
 localparam AHBW = (XLEN);
@@ -213,6 +221,7 @@ localparam PLIC_GPIO_ID = 32'd3;
 localparam PLIC_UART_ID = 32'd10;
 localparam PLIC_SPI_ID = 32'd6;
 localparam PLIC_SDC_ID = 32'd9;
+localparam PLIC_HYDRA_ID = 32'd8;
 
 // Branch prediction
 localparam logic BPRED_SUPPORTED = 1;
