diff --git a/src/cvw.sv b/src/cvw.sv
index 64675b0f7..eb3951b9c 100644
--- a/src/cvw.sv
+++ b/src/cvw.sv
@@ -2,7 +2,7 @@
 // cvw.sv
 //
 // Written: David_Harris@hmc.edu 27 January 2022
-//
+// Modified: Giovanni Sirtori <gs86@rice.edu> (HYDRA integration), 4 May 2026
 // Purpose: package with shared CORE-V-Wally global parameters
 //
 // A component of the Wally configurable RISC-V project.
@@ -145,6 +145,13 @@ typedef struct packed {
   logic         SPI_SUPPORTED;
   logic [63:0]  SPI_BASE;
   logic [63:0]  SPI_RANGE;
+  logic         HYDRA_SUPPORTED;
+  logic [63:0]  HYDRA_BASE;
+  logic [63:0]  HYDRA_RANGE;
+  logic [63:0]  HYDRA_MMR_BASE;
+  logic [63:0]  HYDRA_MMR_RANGE;
+  logic [63:0]  HYDRA_SCRATCH_BASE;
+  logic [63:0]  HYDRA_SCRATCH_RANGE;
 
 // Test modes
 
@@ -162,6 +169,7 @@ typedef struct packed {
   int           PLIC_UART_ID;
   int           PLIC_SPI_ID;
   int           PLIC_SDC_ID;
+  int           PLIC_HYDRA_ID;
 
   logic                BPRED_SUPPORTED;
   logic [31:0]         BPRED_TYPE;
