diff --git a/config/shared/parameter-defs.vh b/config/shared/parameter-defs.vh
index 2fc0c0a57..0f0893712 100644
--- a/config/shared/parameter-defs.vh
+++ b/config/shared/parameter-defs.vh
@@ -90,6 +90,13 @@ localparam cvw_t P = '{
   SPI_SUPPORTED :        SPI_SUPPORTED,
   SPI_BASE :        SPI_BASE,
   SPI_RANGE :        SPI_RANGE,
+  HYDRA_SUPPORTED :        HYDRA_SUPPORTED,
+  HYDRA_BASE :        HYDRA_BASE,
+  HYDRA_RANGE :        HYDRA_RANGE,
+  HYDRA_MMR_BASE :        HYDRA_MMR_BASE,
+  HYDRA_MMR_RANGE :        HYDRA_MMR_RANGE,
+  HYDRA_SCRATCH_BASE :        HYDRA_SCRATCH_BASE,
+  HYDRA_SCRATCH_RANGE :        HYDRA_SCRATCH_RANGE,
   GPIO_LOOPBACK_TEST :        GPIO_LOOPBACK_TEST,
   SPI_LOOPBACK_TEST :        SPI_LOOPBACK_TEST,
   UART_PRESCALE :        UART_PRESCALE ,
@@ -99,6 +106,7 @@ localparam cvw_t P = '{
   PLIC_UART_ID :        PLIC_UART_ID,
   PLIC_SPI_ID :        PLIC_SPI_ID,
   PLIC_SDC_ID :        PLIC_SDC_ID,
+  PLIC_HYDRA_ID :        PLIC_HYDRA_ID,
   BPRED_SUPPORTED :        BPRED_SUPPORTED,
   BPRED_TYPE :        BPRED_TYPE,
   BPRED_SIZE :        BPRED_SIZE,
