diff --git a/src/uncore/plic_apb.sv b/src/uncore/plic_apb.sv
index bf95dd51f..de3733280 100644
--- a/src/uncore/plic_apb.sv
+++ b/src/uncore/plic_apb.sv
@@ -2,7 +2,7 @@
 // plic_apb.sv
 //
 // Written: bbracker@hmc.edu 18 January 2021
-// Modified:
+// Modified: Giovanni Sirtori <gs86@rice.edu> (HYDRA integration), 26 Sep 2026
 //
 // Purpose: Platform-Level Interrupt Controller
 //   Based on RISC-V spec (https://github.com/riscv/riscv-plic-spec/blob/master/riscv-plic.adoc)
@@ -50,7 +50,7 @@ module plic_apb import cvw::*;  #(parameter cvw_t P) (
   input  logic                PENABLE,
   output logic [P.XLEN-1:0]   PRDATA,
   output logic                PREADY,
-  input  logic                UARTIntr,GPIOIntr, SPIIntr, SDCIntr,
+  input  logic                UARTIntr,GPIOIntr, SPIIntr, SDCIntr, HYDRAIntr,
   output logic                MExtInt, SExtInt
 );
 
@@ -173,6 +173,7 @@ module plic_apb import cvw::*;  #(parameter cvw_t P) (
     if(P.PLIC_UART_ID != 0) requests[P.PLIC_UART_ID] = UARTIntr;
     if(P.PLIC_SPI_ID != 0) requests[P.PLIC_SPI_ID] = SPIIntr;
     if(P.PLIC_SDC_ID !=0)   requests[P.PLIC_SDC_ID]  = SDCIntr;
+    if(P.PLIC_HYDRA_ID != 0) requests[P.PLIC_HYDRA_ID] = HYDRAIntr;
   end
 
   // pending interrupt request
